backend/src/test/java/com/redhat/cloud/notifications/db/QueryTest.java

```diff
@@ -78,4 +105,295 @@ public void testSort() {
         assertNull(query.getSortBy());
     }
 
+    /**
+     * Tests that no constraint is violated when the {@link Query#offset}, {@link Query#pageNumber} and
+     * {@link Query#pageSize} fields contain a valid value. Reflection is used to grab the values from the annotations
+     * and to generate a valid value that shouldn't trigger any errors.
+     * @throws NoSuchFieldException if the field doesn't exit in the object.
+     * @throws IllegalAccessException if the field could not be updated via reflection.
+     */
+    @Test
+    public void validMinimumFieldsTest() throws NoSuchFieldException, IllegalAccessException {
+        // Get the annotation's values from the "pageSize" field of the "Query" class.
+        final Field pageSizeField = Query.class.getDeclaredField("pageSize");
+        final Max pageSizeMaxAnnotation = pageSizeField.getAnnotation(Max.class);
+        final Min pageSizeMinAnnotation = pageSizeField.getAnnotation(Min.class);
+        final long maxQueryPageSizeValue = pageSizeMaxAnnotation.value();
+        final long minQueryPageSizeValue = pageSizeMinAnnotation.value();
+
+        // Get the annotation's values from the "pageNumber" field of the "Query" class.
+        final Field pageNumberField = Query.class.getDeclaredField("pageNumber");
+        final Min pageNumberMinAnnotation = pageNumberField.getAnnotation(Min.class);
+        final long minQueryPageNumberValue = pageNumberMinAnnotation.value();
+
+        // Get the annotation's values from the "offset" field of the "Query" class.
+        final Field offsetField = Query.class.getDeclaredField("offset");
+        final Min offsetMinAnnotation = offsetField.getAnnotation(Min.class);
+        final long minQueryOffsetValue = offsetMinAnnotation.value();
+
+        // Get a random number that will be used for the validation. The number must be between the "min" and "max"
+        // values of these annotations, so make sure we generate one that satisfies that constraint.
+        final int minRand = (int) Math.min(minQueryOffsetValue, Math.min(minQueryPageNumberValue, minQueryPageSizeValue));
+        final int validValue = ThreadLocalRandom.current().nextInt(minRand, (int) maxQueryPageSizeValue);
```

Comment:
Could we use a fixed (or a set of) number instead of randomly generating one? Will make the test easier to read. You test the edges and some numbers in between

Reply:
Sure, I've added a fixed number instead of a randomly generated one. My idea was to not ever think about these tests again if we change the collection limits, but to be honest I don't think we will ever change them, so... Changed!